test_contextual_lenses/tape_gfp.py

```diff
@@ -0,0 +1,673 @@
+  df['binary_log_f'] = [int(b) for b in df['log_fluorescence'] > 2.5]
+  return df
+
+
+"""Padding."""
+SEQ_LEN = 237
+def pad_seq(seq, pad_char='-'):
+  
+  padded_seq = seq + pad_char*(SEQ_LEN-len(seq))
+  padding_mask = []
+  for _ in range(len(seq)):
+    padding_mask.append([1])
+  for _ in range(SEQ_LEN-len(seq)):
+    padding_mask.append([0])
+  padding_mask = np.array(padding_mask)
+
+  return padded_seq, padding_mask
+
+
+"""Open train data and add one-hots."""
+train_df = dataset_to_df('tape/fluorescence/fluorescence_train.lmdb')
+train_df.primary.apply(len).describe()
+
+train_df['padded_primary'] = train_df.primary.apply(lambda x: pad_seq(x)[0])
+train_df['padding_masks'] = train_df.primary.apply(lambda x: pad_seq(x)[1])
+train_df.padded_primary.apply(len).describe()
+
+AMINO_ACID_VOCABULARY = [
+    'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R',
+    'S', 'T', 'V', 'W', 'Y', '-'
+]
+
+
+def residues_to_one_hot_inds(amino_acid_residues):
+  return np.array([AMINO_ACID_VOCABULARY.index(char) for char in amino_acid_residues])
+
+train_df['one_hot_inds'] = train_df.padded_primary.apply(lambda x: residues_to_one_hot_inds(x))
+
+print(train_df.head())
+
+
+"""Open test data and add one-hots."""
+test_df = dataset_to_df('tape/fluorescence/fluorescence_test.lmdb')
+test_df.primary.apply(len).describe()
+
+test_df['padded_primary'] = test_df.primary.apply(lambda x: pad_seq(x)[0])
+test_df['padding_masks'] = test_df.primary.apply(lambda x: pad_seq(x)[1])
+test_df.padded_primary.apply(len).describe()
+
+test_df['one_hot_inds'] = test_df.padded_primary.apply(lambda x: residues_to_one_hot_inds(x))
+
+print(test_df.head())
+
+
+"""Data batching."""
+def create_train_data(epochs, train_bs=256):
+  train_one_hot_inds = list(train_df['one_hot_inds'].values)
+  N = len(train_one_hot_inds)
+  train_one_hot_inds = tf.data.Dataset.from_tensor_slices(train_one_hot_inds)
+
+  train_fluorescences = train_df['log_fluorescence'].values
+  train_fluorescences = tf.data.Dataset.from_tensor_slices(train_fluorescences)
+
+  train_batches = tf.data.Dataset.zip((train_one_hot_inds, train_fluorescences))
+  train_batches = train_batches.batch(train_bs).shuffle(buffer_size=N, seed=0).repeat(epochs).as_numpy_iterator()
+
+  return train_batches
+
+
+def create_test_data(epochs=1, test_bs=256):
+  test_one_hot_inds = list(test_df['one_hot_inds'].values)
+  test_one_hot_inds = tf.data.Dataset.from_tensor_slices(test_one_hot_inds)
+
+  test_fluorescences = test_df['log_fluorescence'].values
+  test_fluorescences = tf.data.Dataset.from_tensor_slices(test_fluorescences)
+
+  test_batches = tf.data.Dataset.zip((test_one_hot_inds, test_fluorescences))
+  test_batches = test_batches.batch(test_bs).repeat(epochs).as_numpy_iterator()
+
+  return test_batches
+
+test_fluorescences = test_df['log_fluorescence'].values
+
+
+"""Model evaluation."""
+def evaluate(optimizer, test_data, true_fluorescences, plot_title):
```

Comment:
would be good to have this be able to handle any predictor function, so that you could benchmark against sklearn, for example. All you'd have to change is instead of taking an optimizer arg, take a predict_fn arg, and pass predict_fn=optimizer.target below.